### PR TITLE
clean-up cmake build

### PR DESCRIPTION
**Pull request by aerorahul**
This PR is 1 of 2 pull requests to set up a standalone capability of using ncio as an external library (instead of packaging it inside of the GSI).

This PR:
- cleans up the cmake build and uses the standards of other NCEPLIBS.
- removes modulefiles as they don't belong in this repository.
- adds .gitignore
- has no changes to source code.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,69 @@
+list(APPEND fortran_src
+  module_fv3gfs_ncio.f90
```
Comment (CoryMartin-NOAA) [inline src/CMakeLists.txt:2]: Presumably CMake is smart enough to know that there are multiple .f90 files that create this module?

**Review by CoryMartin-NOAA — APPROVED** [review]

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+install(
+  TARGETS ${LIB_TARGETS}
+  EXPORT ${PROJECT_NAME}Exports
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+### Package config
+include(CMakePackageConfigHelpers)
+set(CONFIG_INSTALL_DESTINATION lib/cmake/${PROJECT_NAME})
```
Comment (kgerheiser) [inline src/CMakeLists.txt]: This should also be `CMAKE_INSTALL_LIBDIR`. It's wrong in other libraries too, but may as well fix it here to start with.

**Review by kgerheiser — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (aerorahul) [reply src/CMakeLists.txt:2]: yes. The other `.f90` files are "included" in the `module_fv3gfs_ncio.f90` and cannot be compiled separately.

File: src/CMakeLists.txt (hunk 2 above)
Comment (aerorahul) [reply src/CMakeLists.txt]: done

**Review by kgerheiser — APPROVED** [review]

**Pull request merged**